client/src/components/Sidebar/menu/ActionMenuItem.tsx

```diff
@@ -0,0 +1,111 @@
+import * as React from 'react';
+
+import Tippy from '@tippyjs/react';
+import Icon from '../../Icon/Icon';
+import { MenuItemDefinition, MenuItemProps } from './MenuItem';
+import { gettext } from '../../../utils/gettext';
+import { isDismissed } from '../modules/MainMenu';
+import { WAGTAIL_CONFIG } from '../../../config/wagtailConfig';
+
+export const ActionMenuItem: React.FunctionComponent<
+  MenuItemProps<ActionMenuItemDefinition>
+> = ({ item, slim, path, state, dispatch }) => {
+  const isActive = state.activePath.startsWith(path);
+  const isInSubMenu = path.split('.').length > 2;
+
+  const onClick = (e: React.MouseEvent<HTMLButtonElement>) => {
+    // Do not capture click events with modifier keys or non-main buttons.
+    if (e.ctrlKey || e.shiftKey || e.metaKey || (e.button && e.button !== 0)) {
+      return;
+    }
+
+    if (!isDismissed(item, state)) {
+      dispatch({
+        type: 'set-dismissible-state',
+        item,
+      });
+    }
```

Comment:
Not sure if we'll ever have an `ActionMenuItem` with a dismissible badge, but I tried it and it works quite well.